apps/provider-console/src/components/deployments/DeploymentDetails.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from "react";
+import { Card, CardContent } from "@akashnetwork/ui/components";
+
+import consoleClient from "@src/utils/consoleClient";
+import { getPrettyTimeFromSeconds } from "@src/utils/dateUtils";
+import { findTotalAmountSpentOnLeases, totalDeploymentCost, totalDeploymentTimeLeft } from "@src/utils/deploymentUtils";
+import { formatBytes } from "@src/utils/formatBytes";
+import { uaktToAKT } from "@src/utils/priceUtils";
+
+interface DeploymentDetailsProps {
+  dseq: string;
+  owner: string;
+}
+
+interface DeploymentDetails {
+  id: string;
+  status: string;
+  balance: number;
+  denom: string;
+  createdDate: string;
+  expiryDate: string;
+  timeRemaining: string;
+  totalMonthlyCostUDenom: number;
+  leases: Array<{
+    provider: {
+      address: string;
+      hostUri: string;
+    };
+    status: string;
+    monthlyCostUDenom: number;
+    cpuUnits: number;
+    gpuUnits: number;
+    memoryQuantity: number;
+    storageQuantity: number;
+    gseq: number;
+    oseq: number;
+  }>;
+  amountSpent: number;
+  costPerMonth: number;
+  other: any;
+  timeLeft: number;
+  closedHeight: number | null;
+}
+
+export const DeploymentDetails: React.FC<DeploymentDetailsProps> = ({ dseq, owner }) => {
+  const [deploymentDetails, setDeploymentDetails] = useState<DeploymentDetails | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchDeploymentDetails = async () => {
+      try {
+        setLoading(true);
+        const response: any = await consoleClient.get(`v1/deployment/${owner}/${dseq}`);
```

Comment:
We usually put data fetching in react-query hooks to handle loading state + fetching + cache + separation of concerns.

Reply:
done, added to react-query